### ISSUE TITLE
Tpetra: How should I fillComplete() my BlockCrsMatrix?

### ISSUE THREAD

**Issue by wfspotz** [trilinos/Trilinos#3751]
@trilinos/tpetra 

Attempts to call `fillComplete()` on my `Tpetra::Experimental::BlockCrsMatrix` result in a `has no member named 'fillComplete'` compilation error.  The `getCrsGraph()` method returns a `const` object, implying that the returned graph is not intended to have its `fillComplete()` method called. The documentation for `BlockCrsMatrix` mentions `fillComplete()` throughout, so what exactly am I expected to do?

## Expectations
The `CrsMatrix` and `BlockCrsMatrix` interfaces should be the same wherever the data structures allow. The `fillComplete()` method would seem to be one of these areas.

## Motivation and Context
I have implemented a NOX solver in SPARC that uses `Tpetra::CrsMatrix` as the underlying operator type, and am working to expand this to support the `Tpetra::Experimental::BlockCrsMatrix` operator type. Exactly where and when to call `fillComplete()` within the context of a SPARC solver was an issue, so not having the method at all throws another wrench into the development process.

**Comment by mhoemmen** [comment]
BlockCrsMatrix does not have a fillComplete method.  Here is how you define the BlockCrsMatrix's structure:

  1. Construct a CrsGraph with the desired structure
  2. Call fillComplete on the CrsGraph
  3. Give the CrsGraph to the BlockCrsMatrix's constructor

> The `CrsMatrix` and `BlockCrsMatrix` interfaces should be the same wherever the data structures allow.

BlockCrsMatrix did not exist until 2014.  Sierra Aero requested it with that interface and I implemented it.  I'm not a fan of CrsMatrix's interface (which I didn't design; it was inherited from Epetra and could not be changed by the time I got to it) and have some ideas for improving it.  If you're curious, let's talk more about it offline.  It could be relevant to PyTrilinos' design.
